fabric/Makefile

```diff
@@ -9,64 +9,55 @@ FABRIC_VERSION ?= v2.0.1
 FABRIC_PEER_BIN = $(FABRIC_PATH)/build/bin/peer
 FABRIC_TLCC = $(FABRIC_PATH)/core/scc/tlcc
 
-FABRIC_CURRENT_BRANCH = $(shell cd $(FABRIC_PATH); git rev-parse --abbrev-ref HEAD)
-FABRIC_CURRENT_TAG = $(shell cd $(FABRIC_PATH); git describe --tags)
-FABRIC_PATCHED_BRANCH = $(FABRIC_VERSION)-fpc
+FABRIC_CURRENT_TAG = $(shell cd $(FABRIC_PATH); git tag --points-at HEAD)
 FABRIC_PATCHED = $(FABRIC_PATH)/.fpc_patched
 FABRIC_TRACKED_CHANGES = $(shell cd $(FABRIC_PATH); git status -s | grep -v '\?')
-FABRIC_COMMITS = $(shell cd $(FABRIC_PATH); git log $(FABRIC_VERSION)..HEAD --oneline)
-FABRIC_LOG_DIFF = $(shell diff $(FABRIC_PATCHED) <(FABRIC_COMMITS))
+FABRIC_CURRENT_DIFF_COMMITS_CMD = git log $(FABRIC_VERSION)..HEAD --oneline
 
 build: native
 
 patch: $(FABRIC_PATCHED)
 
 $(FABRIC_PATCHED):
 	@echo "Patching Fabric ..."
-	@if [ "$(FABRIC_CURRENT_BRANCH)" != "$(FABRIC_PATCHED_BRANCH)" ]; then \
-		echo "Aborting! Current branch is $(FABRIC_CURRENT_BRANCH) but want $(FABRIC_PATCHED_BRANCH)"; exit 1; \
-	fi
-
 	@if [ "$(FABRIC_CURRENT_TAG)" != "$(FABRIC_VERSION)" ]; then \
-		echo "Aborting! Current branch HEAD does not match tag/$(FABRIC_VERSION)! $(FABRIC_CURRENT_TAG) but want $(FABRIC_VERSION)"; exit 1; \
+		echo "Aborting! Tag on current HEAD ($(FABRIC_CURRENT_TAG)) does not match expected tag/$(FABRIC_VERSION)!"; exit 1; \
 	fi
 
 	cd $(FABRIC_PATH) && \
 	git am $(FPC_PATH)/fabric/*.patch && \
-	git log $(FABRIC_VERSION)..HEAD --oneline > $@
+	$(FABRIC_CURRENT_DIFF_COMMITS_CMD) > $@
 
 clean-patch:
-	@echo "Cleaning Fabric patches ..."
-	@if [ "$(FABRIC_CURRENT_BRANCH)" != "$(FABRIC_PATCHED_BRANCH)" ]; then \
-		echo "Aborting! Current branch is $(FABRIC_CURRENT_BRANCH) but want $(FABRIC_PATCHED_BRANCH)"; \
-		echo "Please checkout $(FABRIC_PATCHED_BRANCH) and come back."; \
-		exit 1; \
-	fi
-
-	@if [ "$(FABRIC_CURRENT_TAG)" = "$(FABRIC_VERSION)" ]; then \
-		echo "Aborting! Nothing to do here ... it seems you haven't applied the patches yet"; exit 1; \
-	fi
-
-	@if [ ! -z "$(FABRIC_TRACKED_CHANGES)" ]; then \
-		echo "Aborting! It seems you have some changes in your branch:"; \
+	@echo "Cleaning Fabric patches ..."; \
```

Comment:
is this really needed to have almost everything in this target in a single line? we have not established common syntax for writing makefiles but there we more semicolons and backslashes than in other places.

Reply:
yes, actually _everything_ has to be in a single execution or the exit 0 does not work ...

Reply:
Yea, difficult to review. 

My understanding here is that all of this needs to take place within the context of the fabric path, which is difficult since we can't place our own makefile instructions inside to call out to it.
The result here is that everything has to happen in the same shell execution.

Any consideration of this special target?
https://www.gnu.org/software/make/manual/html_node/One-Shell.html

Reply:
I didn't know about `.ONESHELL`.  But it seems to have major drawbacks as it applies to all targets and, more importantly, now you have to do error checks on every but last step or join everything via '&&' ...